File: core/integrations/augmentation_runtime.py
from __future__ import annotations

from typing import Any, Dict

from core.integrations.capability_registry import supports_capability, supports_provider_capability


def augment_metadata(
    bundle: Dict[str, Any],
    capability: str,
    provider_result: Dict[str, Any] | None = None,
    provider: str | None = None,
) -> Dict[str, Any]:
    """LLM/provider output isolated under metadata['llm'] only."""
    meta = dict(bundle.get("metadata", {}) or {})
    llm = dict(meta.get("llm", {}) or {})
    enabled = (
        supports_provider_capability(provider, capability)
        if provider
        else supports_capability(capability)
    )
    if enabled and provider_result:
        llm[capability] = provider_result
    meta["llm"] = llm
    return {**bundle, "metadata": meta}

File: src/integrations/augmentationRuntime.ts
/**
 * Converted from Python: core/integrations/augmentation_runtime.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { supportsCapability, supportsProviderCapability } from "./capabilityRegistry.js";

export function augmentMetadata(bundle: any, capability: any, provider_result: any = null, provider: any = null): any {
  var meta: any = py.pyDict(py.or2(py.get(bundle, "metadata", {}), () => ({})));
  var llm: any = py.pyDict(py.or2(py.get(meta, "llm", {}), () => ({})));
  var enabled: any = (py.truthy(provider) ? supportsProviderCapability(provider, capability) : supportsCapability(capability));
  if ((py.truthy(enabled) && py.truthy(provider_result))) {
    py.setItem(llm, capability, provider_result);
  }
  py.setItem(meta, "llm", llm);
  return {...(bundle), "metadata": meta};
}
export { supportsCapability, supportsProviderCapability };
